packages/node_modules/@ciscospark/plugin-rooms/src/rooms.js

```diff
@@ -24,6 +33,111 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
+   */
+  onWebexApiEvent(event) {
+    const {activity} = event.data;
+
+    /* eslint-disable no-case-declarations */
+    switch (activity.verb) {
+      case SDK_EVENT.FROM_WEBEX.ACTIVITY_VERB.CREATE:
+        const roomCreatedEvent =
+          this.getRoomEvent(activity, SDK_EVENT.TO_APP.EVENT_TYPE.CREATED);
+
+        if (roomCreatedEvent) {
+          debug(`room "created" payload: \
+            ${JSON.stringify(roomCreatedEvent)}`);
+          this.trigger(SDK_EVENT.TO_APP.EVENT_TYPE.CREATED, roomCreatedEvent);
+        }
+        break;
+
+      case SDK_EVENT.FROM_WEBEX.ACTIVITY_VERB.UPDATE:
+        const roomUpdatedEvent =
+          this.getRoomEvent(activity, SDK_EVENT.TO_APP.EVENT_TYPE.UPDATED);
+
+        if (roomUpdatedEvent) {
+          debug(`room "updated" payload: \
+            ${JSON.stringify(roomUpdatedEvent)}`);
+          this.trigger(SDK_EVENT.TO_APP.EVENT_TYPE.UPDATED, roomUpdatedEvent);
+        }
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs the data object for an event on the rooms resource,
+   * adhering to Hydra's Webehook data structure.
+   * @param {Object} activity from mercury
+   * @param {Object} event type of "webhook" event
+   * @returns {Object} constructed event
+   */
+  getRoomEvent(activity, event) {
+    try {
+      const sdkEvent = cloneDeep(this.eventEnvelope);
+      let room;
+
+      sdkEvent.event = event;
+      sdkEvent.data.created = activity.published;
+      sdkEvent.actorId = constructHydraId(hydraTypes.PEOPLE, activity.actor.id);
+      if (event === SDK_EVENT.TO_APP.EVENT_TYPE.CREATED) {
+        room = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.OBJECT;
+        sdkEvent.data.creatorId =
+          constructHydraId(hydraTypes.PEOPLE, activity.actor.entryUUID);
+        sdkEvent.data.lastActivity = activity.published;
+      }
+      else if (event === SDK_EVENT.TO_APP.EVENT_TYPE.UPDATED) {
+        room = SDK_EVENT.FROM_WEBEX.ACTIVITY_FIELD.TARGET;
+        // TODO - lastActivity is not in the update mercury event
+        // would need to make an async call to get more conversation details
+        // or we can skip this for the SDK, since we will provide an
+        // SDK api to allow the app to query this directly if it needs
+        // sdkEvent.data.lastActivity = 'to do';
+      }
+      else {
+        throw new Error('unexpected event type');
+      }
+      sdkEvent.data.id =
+        constructHydraId(hydraTypes.ROOM, activity[room].id);
+      sdkEvent.data.type =
+        activity[room].tags.includes(SDK_EVENT.FROM_WEBEX.ACTIVITY_TAG.ONE_ON_ONE) ?
+          SDK_EVENT.TO_APP.SPACE_TYPE.DIRECT :
+          SDK_EVENT.TO_APP.SPACE_TYPE.GROUP;
+      sdkEvent.data.isLocked =
+        activity[room].tags.includes(SDK_EVENT.FROM_WEBEX.ACTIVITY_TAG.LOCKED);
+
+      return sdkEvent;
+    }
+    catch (e) {
+      console.error(`Unable to generate SDK event from mercury \
```

Comment:
Use `this.spark.logger`.

Reply:
Agree, see commit 41ff1a54e925f04e1588abc4d53b969c05b123cc